plugins/inputs/fritzbox/fritzbox_test.go

```diff
@@ -0,0 +1,95 @@
+package fritzbox
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tdrn-org/go-tr064/mock"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestConfig(t *testing.T) {
+	// Verify plugin can be loaded from config
+	conf := config.NewConfig()
+	require.NoError(t, conf.LoadConfig("testdata/conf/fritzbox.conf"))
+	require.Len(t, conf.Inputs, 1)
+	f, ok := conf.Inputs[0].Input.(*Fritzbox)
+	require.True(t, ok)
+
+	// Verify successful Init
+	require.NoError(t, f.Init())
+
+	// Verify everything is setup according to config file
+	require.ElementsMatch(t, []string{"http://boxuser:boxpassword@fritz.box:49000/", "http://:repeaterpassword@fritz.repeater:49000/"}, f.URLs)
+	require.Equal(t, []string{"device", "wan", "ppp", "dsl", "wlan", "hosts"}, f.Collect)
+	require.Equal(t, config.Duration(60*time.Second), f.Timeout)
+	require.Equal(t, "secret", f.TLSKeyPwd)
+	require.True(t, f.InsecureSkipVerify)
+}
+
+func TestGather(t *testing.T) {
+	// Start mock server
+	tr064Server := mock.Start("testdata/mock",
+		mock.ServiceMockFromFile("/deviceinfo", "testdata/mock/DeviceInfo.xml"),
+		mock.ServiceMockFromFile("/wancommonifconfig", "testdata/mock/WANCommonInterfaceConfig1.xml"),
+		mock.ServiceMockFromFile("/WANCommonIFC1", "testdata/mock/WANCommonInterfaceConfig2.xml"),
+		mock.ServiceMockFromFile("/wanpppconn", "testdata/mock/WANPPPConnection.xml"),
+		mock.ServiceMockFromFile("/wandslifconfig", "testdata/mock/WANDSLInterfaceConfig.xml"),
+		mock.ServiceMockFromFile("/wlanconfig", "testdata/mock/WLANConfiguration.xml"),
+		mock.ServiceMockFromFile("/hosts", "testdata/mock/Hosts.xml"),
+	)
+	defer tr064Server.Shutdown()
```

Comment:
Please move this to the inner test cycle and load the files from the respective test-case. This eases adding new test-cases later based on bug-reports of users as we can simply ask them to post those files and copy them to a new test-case directory.

Reply:
Adapted via a3b25ca0621333f40757fe28df94eda547096b1f